src/main/java/column/store/bench/SpansBench.java

```diff
@@ -0,0 +1,222 @@
+package column.store.bench;
+
+import column.store.api.column.BooleanColumn;
+import column.store.api.column.Column;
+import column.store.api.column.DoubleColumn;
+import column.store.api.column.IdColumn;
+import column.store.api.column.LongColumn;
+import column.store.api.column.StringColumn;
+import column.store.api.query.Query;
+import column.store.api.write.BooleanColumnWriter;
+import column.store.api.write.DoubleColumnWriter;
+import column.store.api.write.IdColumnWriter;
+import column.store.api.write.LongColumnWriter;
+import column.store.api.write.StringColumnWriter;
+import column.store.api.write.Writer;
+import column.store.csv.write.CSVWriter;
+import column.store.inmemory.InMemoryReader;
+import column.store.inmemory.InMemoryReader.BooleanByteReader;
+import column.store.inmemory.InMemoryReader.DoubleByteReader;
+import column.store.inmemory.InMemoryReader.IdByteReader;
+import column.store.inmemory.InMemoryReader.LongByteReader;
+import column.store.inmemory.InMemoryReader.StringByteReader;
+import column.store.parquet.write.ParquetWriter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+
+public final class SpansBench {
+
+  public static void main(final String[] args) throws IOException {
+    var reader = new InMemoryReader();
+    var path = Path.of("..", "column-store-tools", "4a5355be-70fa-4a24-926e-28392ad6a5bc");
+    var query = Query.from(path).select(Column.forBoolean("dummy")).allOf();
+    reader.query(query);
+
+    var columns = reader.columns();
+    var byType = columns.stream().collect(Collectors.groupingBy(Column::type));
+
+    var booleanReaders = booleanReaders(byType, reader);
+    var doubleReaders = doubleReaders(byType, reader);
+    var idReaders = idReaders(byType, reader);
+    var longReaders = longReaders(byType, reader);
+    var stringReaders = stringReaders(byType, reader);
+
+    var writer = parquetWriter(columns);
+//    var writer = csvWriter();
```

Comment:
@Eliasrpx using the CSW writer does not work here, I don't know why. It tries to read a line from the file it writes to. Should this be fixed?